### PR TITLE
Make global_context consistent.

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Avoid copying global_context. Let's see whether it'll break anything.

Detailed description / Documentation draft:

none.

### CODE REVIEW

File: src/Storages/StorageMongoDB.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ class StorageMongoDB final : public ext::shared_ptr_helper<StorageMongoDB>, publ
     std::string username;
     std::string password;
 
-    Context global_context;
```
Comment (tavplubix) [inline src/Storages/StorageMongoDB.h:54]: Seems like it's never used.
@ageraab, why do we need it?

File: src/Storages/StorageBuffer.h [hunk 2]
```diff
@@ -102,7 +102,7 @@ friend class BufferBlockOutputStream;
 
 
 private:
-    Context global_context;
+    Context & global_context;
```
Comment (tavplubix) [inline src/Storages/StorageBuffer.h:105]:
```suggestion
    const Context & global_context;
```

**Review by tavplubix — APPROVED** [review]

File: src/Storages/StorageMongoDB.h (hunk 1 above)
Comment (alexey-milovidov) [reply src/Storages/StorageMongoDB.h:54]: I would not even ask... if it's unused, we don't need it.

File: src/Storages/StorageBuffer.h [hunk 3]
```diff
@@ -102,7 +108,7 @@ friend class BufferBlockOutputStream;
 
 
 private:
-    Context global_context;
```
Comment (azat) [inline src/Storages/StorageBuffer.h:105]: It will be great to use forward declaration for the Context in the similar places

Comment (amosbird) [reply src/Storages/StorageBuffer.h:105]: Yeah. It'll be more consistent. Done.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests - test_storage_kafka - failure is related: https://gist.github.com/tavplubix/26022e8f3fbdb9451068ed5c98aeeb75

**Comment by amosbird** [comment]
> Integration tests - test_storage_kafka - failure is related: https://gist.github.com/tavplubix/26022e8f3fbdb9451068ed5c98aeeb75

Yeah, it's related.

**Pull request merged**